app/javascript/react/components/animals/ReviewForm.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from "react"
+import { Redirect } from "react-router-dom"
+import _ from "lodash"
+import ErrorList from "../ErrorList"
+
+const ReviewForm = props => {
+  const [reviewFields, setReviewFields] = useState({
+    rating: "",
+    title: "",
+    body: ""
+  })
+  const [errors, setErrors] = useState({})
+
+  const handleInputChange = event => {
+    setReviewFields({
+      ...reviewFields,
+      [event.currentTarget.id]: event.currentTarget.value
+    })
+  }
+
+  const validForSubmission = () => {
+    let submitErrors = {}
+
+    const requiredFields = ["rating", "title", "body"]
+
+    requiredFields.forEach(field => {
+      if(reviewFields[field] === "") {
+        submitErrors = {
+          ...submitErrors,
+          [field]: "can't be blank"
+        }
+      }
+    })
+    setErrors(submitErrors)
+    return _.isEmpty(submitErrors)
+  }
+
+  const handleSubmit = event => {
+    event.preventDefault()
+    if (validForSubmission()) {
+      fetch(`/api/v1/animals/${props.animalId}/reviews`, {
```

Comment:
This fetch ultimately needs to change the state in the parent -- rather than doing the fetch in the child and then triggering a get in the parent, you should move this fetch up into the parent and have `handleSubmit` merely validate and pass up a payload.

This is a pattern you'll see a lot -- needing to move functions up to a "god" component that in turn passes its state down to other components as props. This is just want redux was created to simplify, so stay tuned in postgrad!